flowmachine/tests/test_subscriber_subsetting.py

```diff
@@ -47,165 +45,151 @@ def test_events_table_subscriber_ident_substitutions(ident):
-        # of msisdn_from/msisdn_to
-        calculated_subscriber_set = set(df.subscriber)
 
-        self.assertEqual(calculated_subscriber_set, set(self.subscriber_list))
+def test_cdrs_can_be_subset_by_table(
+    subscriber_list_table, get_dataframe, subscriber_list
+):
+    """
+    We can subset CDRs by a table in the database.
+    """
 
-    def test_subset_correct(self):
-        """Test that pushed in subsetting matches .subset result"""
-        su = EventTableSubset(
-            "2016-01-01", "2016-01-03", subscriber_subset=self.subscriber_list
-        )
-        subsu = EventTableSubset("2016-01-01", "2016-01-03").subset(
-            "subscriber", self.subscriber_list
-        )
-        self.assertTrue(all(su.get_dataframe() == subsu.get_dataframe()))
-        su = HomeLocation(
-            *[
-                daily_location(d, subscriber_subset=self.subscriber_list)
-                for d in list_of_dates("2016-01-01", "2016-01-07")
-            ]
-        )
-        subsu = HomeLocation(
-            *[daily_location(d) for d in list_of_dates("2016-01-01", "2016-01-03")]
-        ).subset("subscriber", self.subscriber_list)
-        self.assertTrue(all(su.get_dataframe() == subsu.get_dataframe()))
+    su = EventTableSubset(
+        "2016-01-01", "2016-01-03", subscriber_subset=subscriber_list_table
+    )
 
-    def test_query_can_be_subscriber_set_restricted(self):
-        """Test that some queries can be limited to only a subset of subscribers."""
+    df = get_dataframe(su)
 
-        # Create a temporary table in the DB
-        con = Table.connection.engine
+    # Get the set of subscribers present in the dataframe, we need to handle the logic
+    # of msisdn_from/msisdn_to
+    calculated_subscriber_set = set(df.subscriber)
 
-        sql = "DROP TABLE IF EXISTS subscriber_list"
-        con.execute(sql)
+    assert calculated_subscriber_set == set(subscriber_list)
 
-        sql = """CREATE TABLE subscriber_list (subscriber TEXT)"""
-        con.execute(sql)
 
-        formatted_subscribers = ",".join(
-            "('{}')".format(u) for u in self.subscriber_list
-        )
-        sql = """INSERT INTO subscriber_list (subscriber) VALUES {}""".format(
-            formatted_subscribers
-        )
-        con.execute(sql)
-        rog = RadiusOfGyration(
-            "2016-01-01", "2016-01-03", subscriber_subset=Table("subscriber_list")
-        )
-        hl = HomeLocation(
-            *[
-                daily_location(d, subscriber_subset=Table("subscriber_list"))
-                for d in list_of_dates("2016-01-01", "2016-01-03")
-            ]
-        )
-        rog_df = rog.get_dataframe()
-        hl_df = hl.get_dataframe()
-        sql = "DROP TABLE IF EXISTS subscriber_list"
-        con.execute(sql)
+def test_subset_correct(subscriber_list, get_dataframe):
+    """Test that pushed in subsetting matches .subset result"""
```

Comment:
What does "pushed in" mean here?

Reply:
Pushed in, in the sense that the subset is shoved down to the initial select from an events table, rather than being done on the home location.